src/web-auth/index.js

```diff
@@ -182,8 +182,11 @@ WebAuth.prototype.validateAuthenticationResponse = function(options, parsedHash,
   var state = parsedHash.state;
   var transaction = this.transactionManager.getStoredTransaction(state);
   var transactionState = options.state || (transaction && transaction.state) || null;
+
   var transactionStateMatchesState = transactionState === state;
-  if (!state || !transactionStateMatchesState) {
+  var shouldBypassStateChecking = !state && !transactionState && options.__enableImpersonation;
```

Comment:
`state` and `transactionState` are missing for *every* impersonation call? 
Also I don't know how this boolean checks works on JS but probably `options. __enableImpersonation` will be false 99% of the cases, so I'd move that check to the beginning of the line.

Reply:
state might be present if you used the advanced fields in the page: 
![image](https://user-images.githubusercontent.com/941075/37171222-ee23d836-22eb-11e8-9e4d-1e60ff167c66.png)

`var transactionState = options.state || (transaction && transaction.state) || null;`
transactionState will be either provided by the developer (`options.state`) or empty when you're trying to impersonate

Reply:
So if the user provides a value in that field (image you shared). Where does that end up being added, options.state (transaction state) or the request state?